framework/contrib/hit/parse.cc

```diff
@@ -921,8 +921,13 @@ parse(const std::string & fname, const std::string & input)
     throw ParseError(input_errors.str());
 
   std::unique_ptr<Node> root(new Section(interpreter, interpreter->root()));
-  std::size_t starting_line = 0;
-  buildHITTree(interpreter, interpreter->root(), root.get(), starting_line);
+
+  if (!interpreter->root().is_null())
```

Comment:
Good point, that's definitely not clear if someone just looks at the code without a comment there.

The `interpreter->root().is_null()` condition is true when `hit::parse()` is given an empty input string.

This actually happens in practice every time MOOSE is ran without any command line parameters provided.

The `framework/src/parser/Parser.C` code that calls `hit::parse()` is structured like this outlined version.

```cpp
void Parser::parse()
{
  for (auto & input_filename : _input_filenames)
  {
    std::unique_ptr<hit::Node> root(hit::parse(input_filename, input));
    hit::merge(root.get(), _root.get());
  }
  auto cli_input = hitCLIFilter(_app.name(), _app.commandLine()->getArguments());
  _cli_root.reset(hit::parse("CLI_ARGS", cli_input));
  hit::merge(_cli_root.get(), _root.get());

  // . . . now use _root merge of input files and command line parameters . . .
}
```

When no command line parameters are given, that second `hit::parse()` is still called with any empty string.

It's been set up like that since before I was involved, so the first line of my `buildHITTree()` has always been:

```cpp
void buildHITTree()
{
  if (hnv_parent.is_null())
    return;

  // . . . build the tree otherwise . . .
}
```

I just added the `is_null()` check here so that `buildHITTree()` is never even called if it will just be a no-op.

I'll add a comment explaining the condition prevents calling the HIT node tree build method with empty input.